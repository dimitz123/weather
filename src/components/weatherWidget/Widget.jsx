import React from 'react';
import PropTypes from 'prop-types';
import Card from '../shared/Card';

const WindInformation = ({ speed, direction }) => (
  <p className="caption">
    <b className="right-spaced">Wind</b>
    {`${direction} ${speed}km/h`}
  </p>
);

WindInformation.propTypes = {
  direction: PropTypes.string.isRequired,
  speed: PropTypes.string.isRequired,
};

const Widget = ({
  title, units, showWind, data,
}) => (
  <Card>
    <div className="weather-widget">
      <h3>{title}</h3>
      <div className="widget-container">
        <div className="widget-icon">
          <img src={data.iconLocation} alt="weather conditions" />
        </div>
        <div className="widget-data">
          <p>{data.location}</p>
          <p className="highlight">
            {units === 'cel' ? data.celTemp : data.farTemp}
            Â°
          </p>
          {showWind && <WindInformation speed={data.windSpeed} direction={data.windDirection} />}
        </div>
      </div>
    </div>
  </Card>
);

Widget.propTypes = {
  title: PropTypes.string.isRequired,
  units: PropTypes.string.isRequired,
  showWind: PropTypes.bool.isRequired,
  data: PropTypes.shape({
    location: PropTypes.string,
    windDirection: PropTypes.string,
    windSpeed: PropTypes.string,
    iconLocation: PropTypes.string,
    celTemp: PropTypes.string,
    farTemp: PropTypes.string,
  }).isRequired,
};

export default Widget;
