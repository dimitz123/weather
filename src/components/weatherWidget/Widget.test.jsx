import React from 'react';
import renderer from 'react-test-renderer';
import Widget from './Widget';

const tempData = {
  location: 'Sydney',
  windDirection: 'NE',
  windSpeed: '10',
  iconLocation: 'http://openweathermap.org/img/wn/01d@2x.png',
  celTemp: '20',
  farTemp: '65',
};

it('correctly displays temperature in Celcius', () => {
  const component = renderer.create(
    <Widget title="Title of Widget" units="cel" showWind data={tempData} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('correctly displays temperature in Fahrenheit', () => {
  const component = renderer.create(
    <Widget title="Title of Widget" units="far" showWind data={tempData} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('correctly displays a different title', () => {
  const component = renderer.create(
    <Widget title="New title" units="cel" showWind data={tempData} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('correctly displays without wind information', () => {
  const component = renderer.create(
    <Widget title="Title of Widget" units="cel" showWind={false} data={tempData} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
