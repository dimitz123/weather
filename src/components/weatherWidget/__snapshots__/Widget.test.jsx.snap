// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget correctly displays a different title 1`] = `
<div
  className="card"
>
  <div
    className="weather-widget"
  >
    <h3>
      New title
    </h3>
    <div
      className="widget-container"
    >
      <div
        className="widget-icon"
      >
        <img
          alt="weather conditions"
          src="http://openweathermap.org/img/wn/01d@2x.png"
        />
      </div>
      <div
        className="widget-data"
      >
        <p>
          Sydney
        </p>
        <p
          className="highlight"
        >
          20
          째
        </p>
        <p
          className="caption"
        >
          <b
            className="right-spaced"
          >
            Wind
          </b>
          NE 10km/h
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Widget correctly displays temperature in Celcius 1`] = `
<div
  className="card"
>
  <div
    className="weather-widget"
  >
    <h3>
      Title of Widget
    </h3>
    <div
      className="widget-container"
    >
      <div
        className="widget-icon"
      >
        <img
          alt="weather conditions"
          src="http://openweathermap.org/img/wn/01d@2x.png"
        />
      </div>
      <div
        className="widget-data"
      >
        <p>
          Sydney
        </p>
        <p
          className="highlight"
        >
          20
          째
        </p>
        <p
          className="caption"
        >
          <b
            className="right-spaced"
          >
            Wind
          </b>
          NE 10km/h
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Widget correctly displays temperature in Fahrenheit 1`] = `
<div
  className="card"
>
  <div
    className="weather-widget"
  >
    <h3>
      Title of Widget
    </h3>
    <div
      className="widget-container"
    >
      <div
        className="widget-icon"
      >
        <img
          alt="weather conditions"
          src="http://openweathermap.org/img/wn/01d@2x.png"
        />
      </div>
      <div
        className="widget-data"
      >
        <p>
          Sydney
        </p>
        <p
          className="highlight"
        >
          65
          째
        </p>
        <p
          className="caption"
        >
          <b
            className="right-spaced"
          >
            Wind
          </b>
          NE 10km/h
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Widget correctly displays without wind information 1`] = `
<div
  className="card"
>
  <div
    className="weather-widget"
  >
    <h3>
      Title of Widget
    </h3>
    <div
      className="widget-container"
    >
      <div
        className="widget-icon"
      >
        <img
          alt="weather conditions"
          src="http://openweathermap.org/img/wn/01d@2x.png"
        />
      </div>
      <div
        className="widget-data"
      >
        <p>
          Sydney
        </p>
        <p
          className="highlight"
        >
          20
          째
        </p>
      </div>
    </div>
  </div>
</div>
`;
